hack/generator/spec/versioning.md

```diff
@@ -0,0 +1,308 @@
+# Versioning
+
+Specification for how storage versioning will operate for code generated CRD definitions.
+
+We're generating a large volume of CRD definitions based on the JSON schema definitions available for Azure Resource Manager use.
+
+## Goals
+
+**Principle of Least Surprise:** The goal of the service operator is to allow users to consumer Azure resources without having to leave the tooling they are familiar with. We therefore want to do things in the idiomatic Kubernetes fashion, so that they don't experience any nasty surprises.
+
+**Auto-generated conversions:** As far as practical, we want to autogenerating the schema for storage use along with the conversions to and from the actual ARM API versions. Hand coding all the required conversions doesn't scale across all the different Azure sevices, especially with the ongoing rate of change. 
```

Comment:
"autogenerate" instead of "autogenerating".

Reply:
Fixed.